audit-service/src/unit-tests/java/uk/gov/gchq/palisade/service/audit/model/MessageSerialisationTest.java

```diff
@@ -23,40 +23,52 @@
-
 import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static uk.gov.gchq.palisade.service.audit.ApplicationTestData.auditErrorMessage;
+import static uk.gov.gchq.palisade.service.audit.ApplicationTestData.auditSuccessMessage;
+import static uk.gov.gchq.palisade.service.audit.service.ServiceName.DATA_SERVICE;
+import static uk.gov.gchq.palisade.service.audit.service.ServiceName.FILTERED_RESOURCE_SERVICE;
+import static uk.gov.gchq.palisade.service.audit.service.ServiceName.POLICY_SERVICE;
+import static uk.gov.gchq.palisade.service.audit.service.ServiceName.RESOURCE_SERVICE;
+import static uk.gov.gchq.palisade.service.audit.service.ServiceName.USER_SERVICE;
 
 class MessageSerialisationTest {
+
     private static final ObjectMapper MAPPER = new ObjectMapper();
 
     static class MessageTypeSource implements ArgumentsProvider {
         @Override
         public Stream<? extends Arguments> provideArguments(final ExtensionContext extensionContext) {
             return Stream.of(
-                    Arguments.of(ApplicationTestData.auditErrorMessage(ServiceName.USER_SERVICE.name())),
-                    Arguments.of(ApplicationTestData.auditErrorMessage(ServiceName.RESOURCE_SERVICE.name())),
-                    Arguments.of(ApplicationTestData.auditErrorMessage(ServiceName.POLICY_SERVICE.name())),
-                    Arguments.of(ApplicationTestData.auditSuccessMessage(ServiceName.DATA_SERVICE.name())),
-                    Arguments.of(ApplicationTestData.auditSuccessMessage(ServiceName.FILTERED_RESOURCE_SERVICE.name())),
-                    Arguments.of(ApplicationTestData.auditSuccessMessage(ServiceName.USER_SERVICE.name()))
+                arguments(auditErrorMessage(USER_SERVICE)),
+                arguments(auditErrorMessage(RESOURCE_SERVICE)),
+                arguments(auditErrorMessage(POLICY_SERVICE)),
+                arguments(auditSuccessMessage(DATA_SERVICE)),
+                arguments(auditSuccessMessage(FILTERED_RESOURCE_SERVICE)),
+                arguments(auditSuccessMessage(USER_SERVICE))
             );
         }
     }
 
     @ParameterizedTest
     @ArgumentsSource(MessageTypeSource.class)
-    <T> void testSerialiseDeserialiseIsConsistent(final T message) throws JsonProcessingException {
-        // Given some test data
+    void testSerialiseDeserialiseIsConsistent(final Object expectedMessage) throws JsonProcessingException {
+
+        var type = expectedMessage.getClass();
+        var actualMessage = MAPPER.readValue(MAPPER.writeValueAsString(expectedMessage), type);
+
+        assertThat(actualMessage)
+            .as("check deserialised %s is the same as the original using recursive comparison", type)
+            .usingRecursiveComparison()
+            .ignoringFieldsOfTypes(Throwable.class)
+            .isEqualTo(expectedMessage);
```

Comment:
why create a test class for the other Seralising/Deseralising tests if you're not using it here also?

Reply:
Too true. Much better. Sorted.